### DISCUSSION TITLE
Navigate to custom url after connection established

### DISCUSSION THREAD

**Discussion by hjkoster400d** [Hieromon/AutoConnect#523] [Feature Request]
Hi there,

Thank you for sharing your awesome repo.

I'm using it to onboard devices onto the network. 
I'm serving a html on the same ESP (PROGMEM page and bound to the server using the on() method), and i would like to navigate to that uri after successful connecting to the wifi. At this moment the page is showing the ESP status (/_ac) page after success.

1. Boot
2. Connect to SoftAP with my phone
3. Enter credentials for wifi connection
4. Wait for success
5. Navigate to /onboarding path

Is there an easy way to accomplish this?

**Comment by Hieromon** [comment]
@hjkoster400d Thank you for your interest in AutoConnect.

Unfortunately, the current AutoConnect API is equipped with does not allow assigning anything other than the embedded page as response page content after the WiFi connection is established. It can only change the URL; the content remains the same as the current.

However, it is possible to support it. In doing so, we have two options for its API, as follows:

1. Determined by preprocessor macros at build time.

    ```cpp
    #define AUTOCONNECT_URI_ONSUCCESS "/onboarding"
    ```
    ```cpp
    WebServer server;
    AutoConnect portal(server);

    void setup() {
      ...
      server.on("/onboarding", []() {
        server.send(200, "text/plain", "WiFi connected");
      });
      portal.config(config);
      portal.begin();
    }
    ```
    The above directive will be defined in the library resource file: `AutoConnectDefs.h`. This value can be changed in the `AutoConnectDefs.h` file and must be modified directly by each build time.
    If PlatformIO is used as the build system, the following can be described in the `platformio.ini` file without changing the `AutoConnectDefs.h` file.
    ```ini
    build_flags = -DAUTOCONNECT_URL_ONSUCCESS='"/onboarding"'
    ```

2. Add a new setting to specify the *OnConnectSuccess* URL with `AutoConnectConfig`

    ```cpp
    WebServer server;
    AutoConnect portal(server);
    AutoConnectConfig config;

    void setup() {
      ...
      server.on("/onboarding", []() {
        server.send(200, "text/plain", "WiFi connected");
      });
      config.successUri = "/onboarding";
      portal.config(config);
      portal.begin();
    }
    ```
    ```cpp
    String AutoConnectConfig::successUri
    ```

Personally, I prefer preprocessor macro directives. I don't think there are many use cases where we need this feature and dynamically replace content at runtime.
How about you?

**Reply by hjkoster400d** [reply]
Hi,

Thanks for the detailed response, i will try option 1 and check if it works for our use case. 
I will let you know.
